### ISSUE TITLE
issue with the almblm form from Wien2k 19.1

### ISSUE THREAD

**Issue by pokornyv** [TRIQS/dft_tools#127]
Dear all,
  I just started to learn how to use DFTTools on the simple SrVO3 example from the tutorial, but I am running into issues. It seems they come from the incompatibility with the Wien2k version.

I use the docker://flatironinstitute/triqs image for DFTTools and Wien2k-19.1 (compiled using ifort from xe 2019.1.053).

### Description

First issue comes from the `case.cf_d_eg_t2g` file from Wien2k `SRC_templates`. It begins as

```
# eg t2g
```

and I get the error 

`At line 140 of file /src/triqs_dft_tools/fortran/dmftproj/set_ang_trans.f
Fortran runtime error: Bad real number in item 1 of list input`

Removing the comment from the file solves that issue (this was recently pointed out by @cmtcl18 in issue #122, but closed w/o solution).

But then I run in a new issue with the `case.almblm` file:

`At line 541 of file /src/triqs_dft_tools/fortran/dmftproj/dmftproj.f (unit = 9, file = 'SrVO3_gga.almblm')
Fortran runtime error: Bad real number in item 1 of list input`

The head of the `SrVO3_gga.almblm` generated by `x lapw2 -almd` goes like

```
   40.9999999999000     
         165
           3
  0.627286660884151     
  6.959499030229120E-002
           1
  0.343272131895234      -0.237351926546292     
 IK =            1  jatom=            1
         497          56          1
           1           1          56
  4.882812500000000E-004  -4.21439675875235     
  4.882812500000000E-004  -2.29006515606318     
  4.882812500000000E-004  -2.29006515606318     
...

```

I opened the `dmftproj.f` code and it seems it assumes the `case.almblm` file in different format.

Is there a simple workaround this issue?

### Steps to Reproduce

1. Run Wien2k-19.1 for paramagnetic SrVO3
2. Copy `case.indmftpr` file from tutorial
3. Run `x lapw2 -almd`
4. Remove header from `case.cf_d_eg_t2g`
5. Run `dmftproj`

### Versions

- OS: Scientific Linux 7.6
- DFTTools from docker://flatironinstitute/triqs (latest)
- Wien2k-19.1

Best.

**Comment by mzingl** [comment]
Hi,
can you check whether the order of your atoms in the struct file is the same as in the .indmftpr file?
I've seen a similar issue recently.
Best, Manuel

**Comment by pokornyv** [comment]
Hi Manuel.
  Thank you for the quick response.

  The `case.struct` file reads (symmetries removed)
```SrVO3                                                                          
P   LATTICE,NONEQUIV.ATOMS:  3 221_Pm-3m                                       
MODE OF CALC=RELA unit=ang                                                     
  7.260331  7.260331  7.260331 90.000000 90.000000 90.000000                   
ATOM   1: X=0.00000000 Y=0.00000000 Z=0.00000000
          MULT= 1          ISPLIT= 2
Sr         NPT=  781  R0=0.00001000 RMT= 2.50000     Z: 38.000                 
LOCAL ROT MATRIX:    1.0000000 0.0000000 0.0000000
                     0.0000000 1.0000000 0.0000000
                     0.0000000 0.0000000 1.0000000
ATOM   2: X=0.50000000 Y=0.50000000 Z=0.50000000
          MULT= 1          ISPLIT= 2
V          NPT=  781  R0=0.00005000 RMT= 1.89        Z: 23.000                 
LOCAL ROT MATRIX:    1.0000000 0.0000000 0.0000000
                     0.0000000 1.0000000 0.0000000
                     0.0000000 0.0000000 1.0000000
ATOM  -3: X=0.00000000 Y=0.50000000 Z=0.50000000
          MULT= 3          ISPLIT=-2
      -3: X=0.50000000 Y=0.00000000 Z=0.50000000
      -3: X=0.50000000 Y=0.50000000 Z=0.00000000
O          NPT=  781  R0=0.00010000 RMT= 1.71        Z:  8.000                 
LOCAL ROT MATRIX:    0.0000000 0.0000000 1.0000000
                     0.0000000 1.0000000 0.0000000
                    -1.0000000 0.0000000 0.0000000
  48      NUMBER OF SYMMETRY OPERATIONS
...
```
and the `case.indmftpr` file:

```
3                ! Nsort
1 1 3            ! Mult(Nsort)
3                ! lmax
complex          ! choice of angular harmonics
1 0 0 0          ! l included for each sort
0 0 0 0          ! If split into ireps, gives number of ireps. for a given orbital (otherwise 0)
cubic            ! choice of angular harmonics
1 1 2 0          ! l included for each sort
0 0 2 0          ! If split into ireps, gives number of ireps. for a given orbital (otherwise 0)
01               !
0                ! SO flag
complex          ! choice of angular harmonics
1 1 0 0          ! l included for each sort
0 0 0 0          ! If split into ireps, gives number of ireps. for a given orbital (otherwise 0)
-0.11 0.14       ! energy window
0.6272866609     ! Fermi energy
```
In both cases the order is Sr - V - O

Best, Vlado.

**Comment by aichhorn** [comment]
Hi,
I could reproduce your problem here. I did the exact same calculation using Wien2k version 14.2 and 19.1, and the two outputs in the almblm file indeed differ. There was some major rewriting of lapw2 in the new Wien2k version, so I assume that this issue is related to that. We need to take a look.

**Comment by aichhorn** [comment]
Seems to be a bug/problem in Wien2k. I checked that in 19.1 some lines are not printed. In the beginning of the almblm file we assume that the overlaps of wave functions are listed for l=0,1,2,3. For 19.1 there is only the overlap for l=0.

**Comment by pokornyv** [comment]
Hi,
  I just installed Wien2k-18.2 and everything works out-of-the-box. `case.cf_d_eg_t2g` does not contain any header and `case.almblm` looks different:

```
   40.9999999999000     
         165
           3
  0.626786066117208     
  6.959758118849396E-002
           1
  0.343462141775549      -0.237369370031787     
  0.242494744504266     
           1
  0.709958523612724      -0.345074057207224     
  5.011768797489954E-002
           0
  9.085655031080634E-003
           0
 IK =            1  jatom=            1
         497          56          1
           1           1          56
  4.882812500000000E-004  -4.21519452851448     
  4.882812500000000E-004  -2.29079329486995     
...
```

So I will stick with Wien2k-18.2 for now.

Thank you for the help.

**Comment by mzingl** [comment]
Thank you for this check!
@aichhorn Can you coordinate that this gets fixed on the Wien2k side?
Best, Manuel

**Comment by aichhorn** [comment]
This is indeed due to a bug in Wien2k that was introduced when restructuring the output of lapw2. According to the Wien2k developers, there is a quite easy fix that will also be implemented in the new Wien2k version. You can find the thread here:

https://www.mail-archive.com/wien@zeus.theochem.tuwien.ac.at/msg19011.html

The patch works as follows. In the SRC_lapw2 directory you open the qdmft.F module file and add
INTEGER :: lmax_to_dmft = 3 
to the very beginning of this module file. Then open l2main.F and comment line 228. Now you recompile lapw2 (be sure to do make clean before) and the case.almblm file looks as it should be. I tested it right now here on my installation and it indeed works.

**Comment by cmtcl18** [comment]
Dear Aichhorn,
I am using WIEN2k - 19.1, I followed the procedure you have mentioned above. But while recompiling lapw2, I am getting error. 

ALMDPrint_tmp_.F:(.text+0x56a): undefined reference to `qdmft_mp_lmax_to_dmft_'
ALMDPrint_tmp_.F:(.text+0x11a1): undefined reference to `qdmft_mp_lmax_to_dmft_'
ALMDPrint_tmp_.F:(.text+0x1c1c): undefined reference to `qdmft_mp_lmax_to_dmft_'
Makefile:179: recipe for target 'lapw2c_mpi' failed
make[1]: *** [lapw2c_mpi] Error 1
make[1]: Leaving directory '/home/hp/WIEN2k_install/SRC_lapw2'
Makefile:164: recipe for target 'cp' failed
make: *** [cp] Error 2
Copying programs
  SRC_lapw2/lapw2
  SRC_lapw2/lapw2c

done.

Compile time errors (if any) were:
SRC_lapw2/compile.msg:make[1]: *** [lapw2_mpi] Error 1
SRC_lapw2/compile.msg:make: *** [rp] Error 2
SRC_lapw2/compile.msg:make[1]: *** [lapw2c_mpi] Error 1
SRC_lapw2/compile.msg:make: *** [cp] Error 2


Check file   compile.msg   in the corresponding SRC_* directory for the 
compilation log and more info on any compilation problem.

Can you suggest, what is the error and where do I need to change anything in the Makefile of SRC_lapw2 directory?

**Comment by aichhorn** [comment]
When putting the variable lmax_to_dmft into the qdmft module, you have to recompile the module. The easiest solution is to do 'make clean' in the SRC_lapw2 directory, before you try to recompile lapw2. This is also mentioned in the thread on the Wien2k mailing list.

As a side note, if you post to a closed issue, you should reopen it!

**Comment by zhl-95** [comment]
> This is indeed due to a bug in Wien2k that was introduced when restructuring the output of lapw2. According to the Wien2k developers, there is a quite easy fix that will also be implemented in the new Wien2k version. You can find the thread here:
> 
> https://www.mail-archive.com/wien@zeus.theochem.tuwien.ac.at/msg19011.html
> 
> The patch works as follows. In the SRC_lapw2 directory you open the qdmft.F module file and add
> INTEGER :: lmax_to_dmft = 3
> to the very beginning of this module file. Then open l2main.F and comment line 228. Now you recompile lapw2 (be sure to do make clean before) and the case.almblm file looks as it should be. I tested it right now here on my installation and it indeed works.

May I ask this solution still available for version 19.2? 
The line 228 in l2main.F reads:
```
227  ! SET cutoff for l printour
228       lmax_to_dmft=3
229       ENDIF
230  ! END QDMFT
```

So if I comment it, there will be a relating error when compiling:

```
qdmft_tmp_.F(2): error #6218: This statement is positioned incorrectly and/or has syntax errors.
MODULE qdmft
^
qdmft_tmp_.F(13): error #6418: This name has already been assigned a data type.   [LMAX_TO_DMFT]
  INTEGER :: nbnd_dmft, lmax_to_dmft
```

Thank you for you reply.
